controllers/base_controller.ts

```diff
@@ -62,12 +59,20 @@ class BaseController<T> {
     async updateItemById(req: Request, res: Response) {
         const itemIdToUpdate = req.params.id;
         const item = req.body;
+        if (Mongoose.prototype.isValidObjectId(itemIdToUpdate)) {
 
-        try {
-            await this.model.findOneAndUpdate({ postId: itemIdToUpdate }, item);
-            res.status(200).send();
-        } catch (error) {
-            res.status(500).send(error);
+            try {
+                const result = await this.model.findByIdAndUpdate(itemIdToUpdate, item, {new: true});
+                if (result !== null) {
```

Comment:
```suggestion
                if (result) {
```
I think it does the same thing

Reply:
Not exactly. Not specifying it means falsly values (zero, undefined...) would be considered false. 
In this case I can switch it though, because we expect to get an object.